feat: expose compute security profile

tamird opened the pull request:
This brings the schema used in `batch_pool_resource` to compute.

Replace top-level booleans with a security_profile block. This will
allow more changes to avoid ForceNew in a later PR. 

- `azurerm_{linux,windows}_virtual_machine{,_scale_set}`: add the 
  `security_profile` block (`security_type`, `host_encryption_enabled`,
  `secure_boot_enabled`, `vtpm_enabled`) and deprecate the legacy
  booleans via `features.DeprecatedInFivePointOh`.

tamird committed:
feat: expose compute security profile

This brings the schema used in `batch_pool_resource` to compute.

Replace top-level booleans with a security_profile block. This will
allow more changes to avoid ForceNew in a later PR.

- `azurerm_{linux,windows}_virtual_machine{,_scale_set}`: add the
  `security_profile` block (`security_type`, `host_encryption_enabled`,
  `secure_boot_enabled`, `vtpm_enabled`) and deprecate the legacy
  booleans via `features.DeprecatedInFivePointOh`.

tamird committed:
[TBS] move old schema behind `!FivePointOh()`

catriona-m commented on internal/services/compute/linux_virtual_machine_resource.go:
```diff
@@ -474,6 +488,41 @@ func resourceLinuxVirtualMachine() *pluginsdk.Resource {
+			Type:       pluginsdk.TypeBool,
+			Optional:   true,
+			ForceNew:   true,
+			Deprecated: features.DeprecatedInFivePointOh("Use `security_profile` block instead."),
```
```suggestion
			Deprecated: features.DeprecatedInFivePointOh("`secure_boot_enabled` has been deprecated in favour of `security_profile.vtpm_enabled` and will be removed in v5.0 of the AzureRM Provider"),
```

tamird replied:
Done.

catriona-m commented on internal/services/compute/linux_virtual_machine_scale_set_resource.go:
```diff
@@ -1447,4 +1447,43 @@ func resourceLinuxVirtualMachineScaleSetSchema() map[string]*pluginsdk.Schema {
+		schema["encryption_at_host_enabled"] = &pluginsdk.Schema{
+			Type:       pluginsdk.TypeBool,
+			Optional:   true,
+			Deprecated: features.DeprecatedInFivePointOh("Use `security_profile` block instead."),
```
could we update this message to be inline with other deprecations?

catriona-m commented on internal/services/compute/linux_virtual_machine_scale_set_resource.go:
```diff
@@ -1447,4 +1447,43 @@ func resourceLinuxVirtualMachineScaleSetSchema() map[string]*pluginsdk.Schema {
+			Type:       pluginsdk.TypeBool,
+			Optional:   true,
+			ForceNew:   true,
+			Deprecated: features.DeprecatedInFivePointOh("Use `security_profile` block instead."),
```
could we update this message to be inline with other deprecations?

tamird replied:
Done.

catriona-m commented on internal/services/compute/virtual_machine_scale_set.go:
```diff
@@ -2222,3 +2224,87 @@ func flattenOrchestratedVirtualMachineScaleSetIdentity(input *identity.SystemAnd
+
+	values := valuesFromVirtualMachineScaleSetSecurityProfile(profile)
+	return securityprofile.ToBlock(values)
+}
```
could we change this to be in line with how we do other flattens in the provider?

tamird replied:
can you be more specific? I think this is the same comment as in batch_pool.go -- in other words, the idea here is to share code between the objects that have a common security_profile shape. Maybe you can tell me what you had in mind?

catriona-m reviewed (changes requested):
HI @tamird thanks for submitting this. I had a look through and left some comments inline. In addition to these, all deprecations will need to be documented in the 5.0 guide ( see the [contributing docs](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/topics/guide-breaking-changes.md#breaking-schema-changes-and-deprecations) for more details, and the documentation for each of the resources will need to be updated to reflect the new properties. Thanks!

tamird commented on internal/services/batch/batch_pool.go:
```diff
@@ -820,32 +819,19 @@ func expandBatchPoolVirtualMachineConfig(d *pluginsdk.ResourceData) (*pool.Virtu
+		UefiSettings: &pool.UefiSettings{
+			SecureBootEnabled: values.SecureBoot,
+			VTpmEnabled:       values.VTPM,
+		},
```
Of course we could, but the idea was to share code between the resources. Can you help me understand why duplication is preferable?

catriona-m replied:
We don't use this pattern anywhere else in the provider, so for consistency and to make this easier to maintain, we should revert back to the original. For this to work correctly it also needs to use casting which could be fragile if it changes in future

tamird replied:
Done in 23945a14bb: removed the shared `ToBlock`/`FromBlock` layer and its casts, restored the original Batch expand/flatten, and kept the compute security-profile conversions local to the VM/VMSS SDK types.

tamird commented on internal/services/compute/virtual_machine_scale_set.go:
```diff
@@ -2222,3 +2224,87 @@ func flattenOrchestratedVirtualMachineScaleSetIdentity(input *identity.SystemAnd
+		}
+
+		return values
+	}()
```
can you be more specific? I think this is the same comment as in batch_pool.go -- in other words, the idea here is to share code between the objects that have a common security_profile shape. Maybe you can tell me what you had in mind?

tamird committed:
[TBS] inline single-use functions

tamird commented on internal/services/compute/linux_virtual_machine_scale_set_resource.go:
```diff
@@ -1045,27 +1022,37 @@ func resourceLinuxVirtualMachineScaleSetRead(d *pluginsdk.ResourceData, meta int
+							}
+							if uefi.SecureBootEnabled != nil {
+								secureBootEnabled = *uefi.SecureBootEnabled
+							}
```
I'm not sure I understand what you mean; the old properties are set just below (lines 1052-1054).

tamird committed:
[TBS] address code review comments

- update deprecation messages
- add comments on encryption_at_host_enabled logic
- add security_profile.host_encryption_enabled test coverage
- update upgrade guide
- update resource documentation

tamird commented on internal/services/compute/linux_virtual_machine_resource.go:
```diff
@@ -1605,8 +1608,33 @@ func resourceLinuxVirtualMachineUpdate(d *pluginsdk.ResourceData, meta interface
+			return false
+		}
+
+		return resolve(blockOld, scalarOld), resolve(blockNew, scalarNew)
```
Done.

catriona-m reviewed (changes requested):
Hi @tamird I left a comment inline explaining why we should remove the To/From Block in this pr, but once that is addressed I can take another look. Thanks!

tamird committed:
fix: keep security profile conversions local

Remove the cross-service ToBlock/FromBlock adapter and restore the
Batch conversion path. Expand and flatten compute security profiles
against their own SDK types so enum pointers are not cast across
services.

Preserve legacy fallbacks and omit an empty security profile from
default VM and scale-set create requests.

catriona-m reviewed (changes requested):
Thanks for updating this @tamird. 

Could we expand on the tests for each of the resources to add steps to update the properties in the blocks and also a step for removing the block?

Also, since there will be no further 4.x releases, we can remove the !features.FivePointOh paths from this which should simplify this pr a bit.

Once these are addressed I can take another look through this. Thanks!

tamird committed:
fix: simplify VM security profiles for v5

Remove obsolete 4.x security-profile compatibility and update VM and
VMSS acceptance coverage for property changes and block removal.
Preserve imports while preventing removed blocks from reappearing on
refresh, and update the corresponding resource documentation.

tamird committed:
fix: remove unused VM security profile imports

catriona-m commented on internal/services/compute/linux_virtual_machine_resource.go:
```diff
@@ -1610,20 +1559,35 @@ func resourceLinuxVirtualMachineUpdate(d *pluginsdk.ResourceData, meta interface
+			return false
+		}
+
+		return resolve(blockOld), resolve(blockNew)
```
we should be using `d.HasChanges` and an `expandVirtualMachineSecurityProfile` to set the block

tamird replied:
Addressed in b072aad763: all four Linux and Windows VM and scale-set update paths now use d.HasChanges("security_profile") and the existing SDK-specific expander. Removing the block explicitly sends EncryptionAtHost=false so the Azure PATCH actually disables host encryption.

— tamirdex

catriona-m commented on internal/services/compute/linux_virtual_machine_resource_disk_os_test.go:
```diff
@@ -370,6 +370,28 @@ func TestAccLinuxVirtualMachine_diskOSConfidentialVmWithGuestStateOnlySecureBoot
+			),
+		},
+		data.ImportStep(),
+	})
```
we should have extra steps in this test to test updating the updatable property `host_encryption_enabled` in the block and also removing the block. we should also include tests to check that the block can be used without any of the optional properties

tamird replied:
Addressed in b072aad763: this Confidential VM test now updates host encryption false → true → false, with an import after each transition. The existing host-encryption lifecycle test separately checks an empty security profile and then complete removal; those cases use a standard VM because VMGuestStateOnly requires vTPM and cannot have an empty or absent security profile.

— tamirdex

catriona-m commented on internal/services/compute/linux_virtual_machine_scale_set_resource_disk_os_test.go:
```diff
@@ -236,6 +236,28 @@ func TestAccLinuxVirtualMachineScaleSet_disksOSDiskConfidentialVmWithGuestStateO
+			),
+		},
+		data.ImportStep("admin_password"),
+	})
```
this test should also include the additional steps

tamird replied:
Addressed in b072aad763: this Confidential VM scale-set test now updates host encryption false → true → false and imports after every transition. The existing standard scale-set lifecycle test additionally covers an empty security profile and complete removal; a VMGuestStateOnly Confidential VM cannot validly omit its required vTPM profile.

— tamirdex

catriona-m commented on internal/services/compute/linux_virtual_machine_scale_set_resource_disk_os_test.go:
```diff
@@ -736,8 +812,11 @@ resource "azurerm_linux_virtual_machine_scale_set" "test" {
-  vtpm_enabled        = true
-  secure_boot_enabled = true
+  security_profile {
+    security_type       = "ConfidentialVM"
```
this should be omitted here since the original test didn't include this
```suggestion
```

tamird replied:
Removed in b072aad763. This pre-existing scale-set fixture no longer adds a ConfidentialVM security type that the original fixture did not configure.

— tamirdex

catriona-m commented on internal/services/compute/virtual_machine.go:
```diff
@@ -665,3 +665,93 @@ func flattenVirtualMachineGalleryApplication(input *[]virtualmachines.VMGalleryA
+		}
+	} else if secureBoot || vtpm {
+		if securityType == nil {
+			securityType = pointer.To(virtualmachines.SecurityTypesTrustedLaunch)
```
`security_profile.security_type` is an optional property with no default so we shouldn't be setting this here when it's not specified in the config

tamird replied:
Addressed in b072aad763 for both VM and scale-set SDKs: an absent or empty security_type remains unset in the expanders, and the flatteners only emit security_type when Azure actually returns one. Neither path invents a default.

— tamirdex

catriona-m replied:
this does get set to an undocumented default here. I verified this by debugging the `TestAccWindowsVirtualMachine_otherSecureBootEnabled` test which has the following config for `security_profile` and does not specify `security_type`:

```
  security_profile {
    secure_boot_enabled = true
  }
```
Here is what the create payload for security_profile contains:

<img width="1153" height="747" alt="Image" src="https://github.com/user-attachments/assets/8fc722d2-190b-46c7-8696-104780c8f66e" />

and causes this failure:

```
=== RUN   TestAccWindowsVirtualMachine_otherSecureBootEnabled
=== PAUSE TestAccWindowsVirtualMachine_otherSecureBootEnabled
=== CONT  TestAccWindowsVirtualMachine_otherSecureBootEnabled
    testcase.go:163: Step 1/2 error: After applying this test step, the non-refresh plan was not empty.
        stdout:
        
        
        Terraform used the selected providers to generate the following execution
        plan. Resource actions are indicated with the following symbols:
        -/+ destroy and then create replacement
        
        Terraform will perform the following actions:
        
          # azurerm_windows_virtual_machine.test must be replaced
        -/+ resource "azurerm_windows_virtual_machine" "test" {
              ~ allow_extension_operations                             = true -> (known after apply)
              ~ automatic_updates_enabled                              = true -> (known after apply)
              ~ computer_name                                          = "acctestvmsq1xp" -> (known after apply)
              ~ disk_controller_type                                   = "SCSI" -> (known after apply)
              ~ hotpatching_enabled                                    = false -> (known after apply)
              ~ id                                                     = "/subscriptions/1a6092a6-137e-4025-9a7c-ef77f76f2c02/resourceGroups/acctestRG-260724181818680387/providers/Microsoft.Compute/virtualMachines/acctestvmsq1xp" -> (known after apply)
                name                                                   = "acctestvmsq1xp"
              ~ os_managed_disk_id                                     = "/subscriptions/1a6092a6-137e-4025-9a7c-ef77f76f2c02/resourceGroups/acctestRG-260724181818680387/providers/Microsoft.Compute/disks/acctestvmsq1xp_OsDisk_1_36a66a0093844d3d9dcc8c1c16b7b3b8" -> (known after apply)
              ~ patch_assessment_mode                                  = "ImageDefault" -> (known after apply)
              ~ patch_mode                                             = "AutomaticByOS" -> (known after apply)
              ~ private_ip_address                                     = "10.0.2.4" -> (known after apply)
              ~ private_ip_addresses                                   = [
                  - "10.0.2.4",
                ] -> (known after apply)
              ~ provision_vm_agent                                     = true -> (known after apply)
              + public_ip_address                                      = (known after apply)
              ~ public_ip_addresses                                    = [] -> (known after apply)
              ~ virtual_machine_id                                     = "de58ed49-669b-452c-9b18-8f4c5f79a117" -> (known after apply)
              ~ vm_agent_platform_updates_enabled                      = true -> (known after apply)
                # (25 unchanged attributes hidden)
        
              ~ os_disk {
                  ~ disk_size_gb                     = 127 -> (known after apply)
                  ~ id                               = "/subscriptions/1a6092a6-137e-4025-9a7c-ef77f76f2c02/resourceGroups/acctestRG-260724181818680387/providers/Microsoft.Compute/disks/acctestvmsq1xp_OsDisk_1_36a66a0093844d3d9dcc8c1c16b7b3b8" -> (known after apply)
                  ~ name                             = "acctestvmsq1xp_OsDisk_1_36a66a0093844d3d9dcc8c1c16b7b3b8" -> (known after apply)
                    # (6 unchanged attributes hidden)
                }
        
              ~ security_profile {
                  - host_encryption_enabled = false -> null
                  - security_type           = "TrustedLaunch" -> null # forces replacement
                  - vtpm_enabled            = false -> null
                    # (1 unchanged attribute hidden)
                }
        
              ~ termination_notification (known after apply)
        
                # (1 unchanged block hidden)
            }
        
        Plan: 1 to add, 0 to change, 1 to destroy.
```

tamird replied:
Confirmed your nonempty-plan diagnosis. Removed inferred security types from both expanders, explicitly configure TrustedLaunch or ConfidentialVM in the applicable fixtures, and document when each type is required. All four schemas now explicitly default the nested booleans to false, so the other false-to-null changes in your captured plan cannot cause replacement either.

— tamirdex

catriona-m commented on internal/services/compute/windows_virtual_machine_scale_set_resource_disk_os_test.go:
```diff
@@ -236,6 +236,28 @@ func TestAccWindowsVirtualMachineScaleSet_disksOSDiskConfidentialVmWithGuestStat
+		},
+		data.ImportStep("admin_password"),
+	})
+}
```
this test also need additional steps

tamird replied:
Addressed in b072aad763: this Windows Confidential VM scale-set test now updates host encryption false → true → false and imports after every transition. The existing standard Windows scale-set lifecycle test independently exercises an empty security profile followed by complete removal, which would be invalid for this VMGuestStateOnly fixture.

— tamirdex

catriona-m commented on internal/services/compute/windows_virtual_machine_scale_set_resource_disk_os_test.go:
```diff
@@ -713,12 +736,65 @@ resource "azurerm_windows_virtual_machine_scale_set" "test" {
-  vtpm_enabled        = %t
-  secure_boot_enabled = %t
+  security_profile {
+    security_type       = "ConfidentialVM"
```
```suggestion
```

tamird replied:
Removed in b072aad763. This existing Windows scale-set fixture no longer introduces a ConfidentialVM security type.

— tamirdex

catriona-m commented on website/docs/r/linux_virtual_machine.html.markdown:
```diff
@@ -356,6 +352,13 @@ A `plan` block supports the following:
+A `security_profile` block supports the following:
+
+* `host_encryption_enabled` - (Optional) Should all of the disks (including the temp disk) attached to this Virtual Machine be encrypted by enabling Encryption at Host?
+* `security_type` - (Optional) Specifies the secure hardware type of this Virtual Machine. Possible values are `Standard`, `TrustedLaunch`, and `ConfidentialVM`. Changing this forces a new resource to be created.
```
the `virtualmachines.PossibleValuesForSecurityTypes()` only returns `TrustedLaunch` and `ConfidentialVM`

```suggestion
* `security_type` - (Optional) Specifies the secure hardware type of this Virtual Machine. Possible values are `TrustedLaunch` and `ConfidentialVM`. Changing this forces a new resource to be created.
```

tamird replied:
Addressed in b072aad763. All four VM and scale-set resource documents now list only the actual SDK-supported security types: TrustedLaunch and ConfidentialVM.

— tamirdex

catriona-m commented on website/docs/r/linux_virtual_machine_scale_set.html.markdown:
```diff
@@ -568,6 +564,13 @@ A `rolling_upgrade_policy` block supports the following:
+A `security_profile` block supports the following:
+
+* `host_encryption_enabled` - (Optional) Should all of the disks (including the temp disk) attached to this Virtual Machine be encrypted by enabling Encryption at Host?
+* `security_type` - (Optional) Specifies the secure hardware type of this Virtual Machine. Possible values are `Standard`, `TrustedLaunch`, and `ConfidentialVM`. Changing this forces a new resource to be created.
```
```suggestion
* `security_type` - (Optional) Specifies the secure hardware type of this Virtual Machine. Possible values are `TrustedLaunch` and `ConfidentialVM`. Changing this forces a new resource to be created.
```

tamird replied:
Addressed in b072aad763: the Linux scale-set documentation now lists only TrustedLaunch and ConfidentialVM.

— tamirdex

catriona-m commented on website/docs/r/windows_virtual_machine.html.markdown:
```diff
@@ -360,6 +356,13 @@ A `plan` block supports the following:
+A `security_profile` block supports the following:
+
+* `host_encryption_enabled` - (Optional) Should all of the disks (including the temp disk) attached to this Virtual Machine be encrypted by enabling Encryption at Host?
+* `security_type` - (Optional) Specifies the secure hardware type of this Virtual Machine. Possible values are `Standard`, `TrustedLaunch`, and `ConfidentialVM`. Changing this forces a new resource to be created.
```
```suggestion
* `security_type` - (Optional) Specifies the secure hardware type of this Virtual Machine. Possible values are `TrustedLaunch` and `ConfidentialVM`. Changing this forces a new resource to be created.
```

tamird replied:
Addressed in b072aad763: the Windows VM documentation now lists only TrustedLaunch and ConfidentialVM.

— tamirdex

catriona-m commented on website/docs/r/windows_virtual_machine_scale_set.html.markdown:
```diff
@@ -557,6 +553,13 @@ A `rolling_upgrade_policy` block supports the following:
+A `security_profile` block supports the following:
+
+* `host_encryption_enabled` - (Optional) Should all of the disks (including the temp disk) attached to this Virtual Machine be encrypted by enabling Encryption at Host?
+* `security_type` - (Optional) Specifies the secure hardware type of this Virtual Machine. Possible values are `Standard`, `TrustedLaunch`, and `ConfidentialVM`. Changing this forces a new resource to be created.
```
```suggestion
* `security_type` - (Optional) Specifies the secure hardware type of this Virtual Machine. Possible values are `TrustedLaunch` and `ConfidentialVM`. Changing this forces a new resource to be created.
```

tamird replied:
Addressed in b072aad763: the Windows scale-set documentation now lists only TrustedLaunch and ConfidentialVM.

— tamirdex

catriona-m commented on internal/services/compute/windows_virtual_machine_scale_set_resource_disk_os_test.go:
```diff
@@ -768,8 +844,11 @@ resource "azurerm_windows_virtual_machine_scale_set" "test" {
-  vtpm_enabled        = true
-  secure_boot_enabled = true
+  security_profile {
+    security_type       = "ConfidentialVM"
```
```suggestion
```

tamird replied:
Removed in b072aad763. The existing Windows scale-set customer-managed-key fixture no longer introduces a ConfidentialVM security type.

— tamirdex

catriona-m commented on internal/services/compute/linux_virtual_machine_scale_set_resource_disk_os_test.go:
```diff
@@ -679,12 +702,65 @@ resource "azurerm_linux_virtual_machine_scale_set" "test" {
-  vtpm_enabled        = %t
-  secure_boot_enabled = %t
+  security_profile {
+    security_type       = "ConfidentialVM"
```
```suggestion
```

tamird replied:
Removed in b072aad763. The existing Linux scale-set customer-managed-key fixture no longer introduces a ConfidentialVM security type.

— tamirdex

catriona-m reviewed (changes requested):
Thanks @tamird I've some comments inline but once those are addressed I can take another look through. Thanks!

tamird committed:
fix: address VM security profile review

catriona-m reviewed (changes requested):
Thanks for updating this @tamird. Unfortunately there are a number of failing tests that will need to be addressed before this can be merged.

```
------- Stdout: -------
=== RUN   TestAccLinuxVirtualMachine_otherEncryptionAtHostEnabledUpdate
=== PAUSE TestAccLinuxVirtualMachine_otherEncryptionAtHostEnabledUpdate
=== CONT  TestAccLinuxVirtualMachine_otherEncryptionAtHostEnabledUpdate
    testcase.go:181: Step 1/5 error: Error running apply: exit status 1
        Error: creating Linux Virtual Machine (Subscription: "*******"
        Resource Group Name: "acctestRG-260724121112490817"
        Virtual Machine Name: "acctestVM-260724121112490817"): performing CreateOrUpdate: unexpected status 400 (400 Bad Request) with error: BadRequest: Use of UEFI settings is not supported for the provided image.
          with azurerm_linux_virtual_machine.test,
          on terraform_plugin_test.tf line 83, in resource "azurerm_linux_virtual_machine" "test":
          83: resource "azurerm_linux_virtual_machine" "test" {
    panic.go:694: Error running post-test destroy, there may be dangling resources: exit status 1
        Error: deleting Network Interface (Subscription: "*******"
        Resource Group Name: "acctestRG-260724121112490817"
        Network Interface Name: "acctestnic-260724121112490817"): performing Delete: unexpected status 400 (400 Bad Request) with error: NicReservedForAnotherVm: Nic(s) in request is reserved for another Virtual Machine for 180 seconds. Please provide another nic(s) or retry after 180 seconds. Reserved VM: /subscriptions/*******/resourceGroups/acctestRG-260724121112490817/providers/Microsoft.Compute/virtualMachines/acctestVM-260724121112490817
--- FAIL: TestAccLinuxVirtualMachine_otherEncryptionAtHostEnabledUpdate (79.71s)
FAIL

------- Stdout: -------
=== RUN   TestAccLinuxVirtualMachine_otherSecureBootEnabled
=== PAUSE TestAccLinuxVirtualMachine_otherSecureBootEnabled
=== CONT  TestAccLinuxVirtualMachine_otherSecureBootEnabled
    testcase.go:181: Step 1/1 error: Error running apply: exit status 1
        Error: creating Linux Virtual Machine (Subscription: "*******"
        Resource Group Name: "acctestRG-260724121223947102"
        Virtual Machine Name: "acctestVM-260724121223947102"): performing CreateOrUpdate: unexpected status 400 (400 Bad Request) with error: BadRequest: Use of UEFI settings is not supported when security type is '<NULL>'.
          with azurerm_linux_virtual_machine.test,
          on terraform_plugin_test.tf line 83, in resource "azurerm_linux_virtual_machine" "test":
          83: resource "azurerm_linux_virtual_machine" "test" {
--- FAIL: TestAccLinuxVirtualMachine_otherSecureBootEnabled (143.20s)
FAIL
    
    
    ------- Stdout: -------
=== RUN   TestAccLinuxVirtualMachineScaleSet_disksOSDiskConfidentialVmWithGuestStateOnlySecureBootEnabled
=== PAUSE TestAccLinuxVirtualMachineScaleSet_disksOSDiskConfidentialVmWithGuestStateOnlySecureBootEnabled
=== CONT  TestAccLinuxVirtualMachineScaleSet_disksOSDiskConfidentialVmWithGuestStateOnlySecureBootEnabled
    testcase.go:181: Step 1/1 error: Error running apply: exit status 1
        Error: creating Linux Virtual Machine Scale Set (Subscription: "*******"
        Resource Group Name: "acctestRG-vmss-260724122924658929"
        Virtual Machine Scale Set Name: "acctestvmss-260724122924658929"): performing CreateOrUpdate: unexpected status 400 (400 Bad Request) with error: BadRequest: The parameter 'managedDisk.securityProfile' can only be set for Confidential VMs.
          with azurerm_linux_virtual_machine_scale_set.test,
          on terraform_plugin_test.tf line 69, in resource "azurerm_linux_virtual_machine_scale_set" "test":
          69: resource "azurerm_linux_virtual_machine_scale_set" "test" {
--- FAIL: TestAccLinuxVirtualMachineScaleSet_disksOSDiskConfidentialVmWithGuestStateOnlySecureBootEnabled (106.47s)
FAIL
    
------- Stdout: -------
=== RUN   TestAccLinuxVirtualMachineScaleSet_disksOSDiskConfidentialSecurityProfileBlock
=== PAUSE TestAccLinuxVirtualMachineScaleSet_disksOSDiskConfidentialSecurityProfileBlock
=== CONT  TestAccLinuxVirtualMachineScaleSet_disksOSDiskConfidentialSecurityProfileBlock
    testcase.go:181: Step 1/3 error: Error running apply: exit status 1
        Error: creating Linux Virtual Machine Scale Set (Subscription: "*******"
        Resource Group Name: "acctestRG-vmss-260724122938534862"
        Virtual Machine Scale Set Name: "acctestvmss-260724122938534862"): performing CreateOrUpdate: unexpected status 400 (400 Bad Request) with error: InvalidParameter: Authentication using either SSH or by user name and password must be enabled in Linux profile.
          with azurerm_linux_virtual_machine_scale_set.test,
          on terraform_plugin_test.tf line 69, in resource "azurerm_linux_virtual_machine_scale_set" "test":
          69: resource "azurerm_linux_virtual_machine_scale_set" "test" {
--- FAIL: TestAccLinuxVirtualMachineScaleSet_disksOSDiskConfidentialSecurityProfileBlock (112.26s)
FAIL
    
------- Stdout: -------
=== RUN   TestAccLinuxVirtualMachineScaleSet_disksOSDiskConfidentialVmWithGuestStateOnlySecureBootDisabled
=== PAUSE TestAccLinuxVirtualMachineScaleSet_disksOSDiskConfidentialVmWithGuestStateOnlySecureBootDisabled
=== CONT  TestAccLinuxVirtualMachineScaleSet_disksOSDiskConfidentialVmWithGuestStateOnlySecureBootDisabled
    testcase.go:181: Step 1/1 error: Error running apply: exit status 1
        Error: creating Linux Virtual Machine Scale Set (Subscription: "*******"
        Resource Group Name: "acctestRG-vmss-260724122937921906"
        Virtual Machine Scale Set Name: "acctestvmss-260724122937921906"): performing CreateOrUpdate: unexpected status 400 (400 Bad Request) with error: BadRequest: The parameter 'managedDisk.securityProfile' can only be set for Confidential VMs.
          with azurerm_linux_virtual_machine_scale_set.test,
          on terraform_plugin_test.tf line 69, in resource "azurerm_linux_virtual_machine_scale_set" "test":
          69: resource "azurerm_linux_virtual_machine_scale_set" "test" {
--- FAIL: TestAccLinuxVirtualMachineScaleSet_disksOSDiskConfidentialVmWithGuestStateOnlySecureBootDisabled (117.44s)
FAIL

------- Stdout: -------
=== RUN   TestAccLinuxVirtualMachineScaleSet_disksOSDiskConfidentialVmWithDiskAndVMGuestStateCMK
=== PAUSE TestAccLinuxVirtualMachineScaleSet_disksOSDiskConfidentialVmWithDiskAndVMGuestStateCMK
=== CONT  TestAccLinuxVirtualMachineScaleSet_disksOSDiskConfidentialVmWithDiskAndVMGuestStateCMK
    testcase.go:181: Step 1/1 error: Error running apply: exit status 1
        Error: creating Linux Virtual Machine Scale Set (Subscription: "*******"
        Resource Group Name: "acctestRG-vmss-260724122948485290"
        Virtual Machine Scale Set Name: "acctestvmss-260724122948485290"): performing CreateOrUpdate: unexpected status 400 (400 Bad Request) with error: BadRequest: The parameter 'managedDisk.securityProfile' can only be set for Confidential VMs.
          with azurerm_linux_virtual_machine_scale_set.test,
          on terraform_plugin_test.tf line 75, in resource "azurerm_linux_virtual_machine_scale_set" "test":
          75: resource "azurerm_linux_virtual_machine_scale_set" "test" {
--- FAIL: TestAccLinuxVirtualMachineScaleSet_disksOSDiskConfidentialVmWithDiskAndVMGuestStateCMK (309.64s)
FAIL
    
------- Stdout: -------
=== RUN   TestAccLinuxVirtualMachineScaleSet_otherEncryptionAtHost
=== PAUSE TestAccLinuxVirtualMachineScaleSet_otherEncryptionAtHost
=== CONT  TestAccLinuxVirtualMachineScaleSet_otherEncryptionAtHost
    testcase.go:181: Step 1/1 error: Error running apply: exit status 1
        Error: creating Linux Virtual Machine Scale Set (Subscription: "*******"
        Resource Group Name: "acctestRG-vmss-260724125031171546"
        Virtual Machine Scale Set Name: "acctestvmss-260724125031171546"): performing CreateOrUpdate: unexpected status 400 (400 Bad Request) with error: BadRequest: Use of UEFI settings is not supported for the provided image.
          with azurerm_linux_virtual_machine_scale_set.test,
          on terraform_plugin_test.tf line 69, in resource "azurerm_linux_virtual_machine_scale_set" "test":
          69: resource "azurerm_linux_virtual_machine_scale_set" "test" {
--- FAIL: TestAccLinuxVirtualMachineScaleSet_otherEncryptionAtHost (103.86s)
FAIL

------- Stdout: -------
=== RUN   TestAccLinuxVirtualMachineScaleSet_otherEncryptionAtHostUpdate
=== PAUSE TestAccLinuxVirtualMachineScaleSet_otherEncryptionAtHostUpdate
=== CONT  TestAccLinuxVirtualMachineScaleSet_otherEncryptionAtHostUpdate
    testcase.go:181: Step 1/5 error: Error running apply: exit status 1
        Error: creating Linux Virtual Machine Scale Set (Subscription: "*******"
        Resource Group Name: "acctestRG-vmss-260724125111318217"
        Virtual Machine Scale Set Name: "acctestvmss-260724125111318217"): performing CreateOrUpdate: unexpected status 400 (400 Bad Request) with error: BadRequest: Use of UEFI settings is not supported for the provided image.
          with azurerm_linux_virtual_machine_scale_set.test,
          on terraform_plugin_test.tf line 69, in resource "azurerm_linux_virtual_machine_scale_set" "test":
          69: resource "azurerm_linux_virtual_machine_scale_set" "test" {
--- FAIL: TestAccLinuxVirtualMachineScaleSet_otherEncryptionAtHostUpdate (103.63s)
FAIL
    
------- Stdout: -------
=== RUN   TestAccLinuxVirtualMachineScaleSet_otherVTpmEnabled
=== PAUSE TestAccLinuxVirtualMachineScaleSet_otherVTpmEnabled
=== CONT  TestAccLinuxVirtualMachineScaleSet_otherVTpmEnabled
    testcase.go:181: Step 1/1 error: Error running apply: exit status 1
        Error: creating Linux Virtual Machine Scale Set (Subscription: "*******"
        Resource Group Name: "acctestRG-vmss-260724125224519307"
        Virtual Machine Scale Set Name: "acctestvmss-260724125224519307"): performing CreateOrUpdate: unexpected status 400 (400 Bad Request) with error: BadRequest: Use of UEFI settings is not supported when security type is '<NULL>'.
          with azurerm_linux_virtual_machine_scale_set.test,
          on terraform_plugin_test.tf line 69, in resource "azurerm_linux_virtual_machine_scale_set" "test":
          69: resource "azurerm_linux_virtual_machine_scale_set" "test" {
--- FAIL: TestAccLinuxVirtualMachineScaleSet_otherVTpmEnabled (119.43s)
FAIL

------- Stdout: -------
=== RUN   TestAccLinuxVirtualMachineScaleSet_otherSecureBootEnabled
=== PAUSE TestAccLinuxVirtualMachineScaleSet_otherSecureBootEnabled
=== CONT  TestAccLinuxVirtualMachineScaleSet_otherSecureBootEnabled
    testcase.go:181: Step 1/1 error: Error running apply: exit status 1
        Error: creating Linux Virtual Machine Scale Set (Subscription: "*******"
        Resource Group Name: "acctestRG-vmss-260724125221751926"
        Virtual Machine Scale Set Name: "acctestvmss-260724125221751926"): performing CreateOrUpdate: unexpected status 400 (400 Bad Request) with error: BadRequest: Use of UEFI settings is not supported when security type is '<NULL>'.
          with azurerm_linux_virtual_machine_scale_set.test,
          on terraform_plugin_test.tf line 69, in resource "azurerm_linux_virtual_machine_scale_set" "test":
          69: resource "azurerm_linux_virtual_machine_scale_set" "test" {
--- FAIL: TestAccLinuxVirtualMachineScaleSet_otherSecureBootEnabled (139.81s)
FAIL

------- Stdout: -------
=== RUN   TestAccLinuxVirtualMachineScaleSet_otherEncryptionAtHostWithCMK
=== PAUSE TestAccLinuxVirtualMachineScaleSet_otherEncryptionAtHostWithCMK
=== CONT  TestAccLinuxVirtualMachineScaleSet_otherEncryptionAtHostWithCMK
    testcase.go:181: Step 1/1 error: Error running apply: exit status 1
        Error: creating Linux Virtual Machine Scale Set (Subscription: "*******"
        Resource Group Name: "acctestRG-260724125156632445"
        Virtual Machine Scale Set Name: "acctestvmss-260724125156632445"): performing CreateOrUpdate: unexpected status 400 (400 Bad Request) with error: BadRequest: Use of UEFI settings is not supported for the provided image.
          with azurerm_linux_virtual_machine_scale_set.test,
          on terraform_plugin_test.tf line 152, in resource "azurerm_linux_virtual_machine_scale_set" "test":
         152: resource "azurerm_linux_virtual_machine_scale_set" "test" {
--- FAIL: TestAccLinuxVirtualMachineScaleSet_otherEncryptionAtHostWithCMK (305.58s)
FAIL

------- Stdout: -------
=== RUN   TestAccWindowsVirtualMachine_diskOSConfidentialVmWithGuestStateOnlySecureBootEnabled
=== PAUSE TestAccWindowsVirtualMachine_diskOSConfidentialVmWithGuestStateOnlySecureBootEnabled
=== CONT  TestAccWindowsVirtualMachine_diskOSConfidentialVmWithGuestStateOnlySecureBootEnabled
    testcase.go:181: Step 1/1 error: Error running apply: exit status 1
        Error: creating Windows Virtual Machine (Subscription: "*******"
        Resource Group Name: "acctestRG-260724130513178723"
        Virtual Machine Name: "acctestvmjx2vf"): performing CreateOrUpdate: unexpected status 404 (404 Not Found) with error: PlatformImageNotFound: The platform image 'MicrosoftWindowsServer:windows-cvm:2022-datacenter-cvm:latest' is not available. Verify that all fields in the storage profile are correct. For more details about storage profile information, please refer to https://aka.ms/storageprofile
          with azurerm_windows_virtual_machine.test,
          on terraform_plugin_test.tf line 78, in resource "azurerm_windows_virtual_machine" "test":
          78: resource "azurerm_windows_virtual_machine" "test" {
--- FAIL: TestAccWindowsVirtualMachine_diskOSConfidentialVmWithGuestStateOnlySecureBootEnabled (122.81s)
FAIL

------- Stdout: -------
=== RUN   TestAccWindowsVirtualMachine_diskOSConfidentialVmWithGuestStateOnlySecureBootDisabled
=== PAUSE TestAccWindowsVirtualMachine_diskOSConfidentialVmWithGuestStateOnlySecureBootDisabled
=== CONT  TestAccWindowsVirtualMachine_diskOSConfidentialVmWithGuestStateOnlySecureBootDisabled
    testcase.go:181: Step 1/1 error: Error running apply: exit status 1
        Error: creating Windows Virtual Machine (Subscription: "*******"
        Resource Group Name: "acctestRG-260724130519136151"
        Virtual Machine Name: "acctestvme2bf6"): performing CreateOrUpdate: unexpected status 404 (404 Not Found) with error: PlatformImageNotFound: The platform image 'MicrosoftWindowsServer:windows-cvm:2022-datacenter-cvm:latest' is not available. Verify that all fields in the storage profile are correct. For more details about storage profile information, please refer to https://aka.ms/storageprofile
          with azurerm_windows_virtual_machine.test,
          on terraform_plugin_test.tf line 78, in resource "azurerm_windows_virtual_machine" "test":
          78: resource "azurerm_windows_virtual_machine" "test" {
--- FAIL: TestAccWindowsVirtualMachine_diskOSConfidentialVmWithGuestStateOnlySecureBootDisabled (122.02s)
FAIL

------- Stdout: -------
=== RUN   TestAccWindowsVirtualMachine_diskOSConfidentialSecurityProfileBlock
=== PAUSE TestAccWindowsVirtualMachine_diskOSConfidentialSecurityProfileBlock
=== CONT  TestAccWindowsVirtualMachine_diskOSConfidentialSecurityProfileBlock
    testcase.go:181: Step 1/3 error: Error running apply: exit status 1
        Error: creating Windows Virtual Machine (Subscription: "*******"
        Resource Group Name: "acctestRG-260724130557499203"
        Virtual Machine Name: "acctestvmjyige"): performing CreateOrUpdate: unexpected status 404 (404 Not Found) with error: PlatformImageNotFound: The platform image 'MicrosoftWindowsServer:windows-cvm:2022-datacenter-cvm:latest' is not available. Verify that all fields in the storage profile are correct. For more details about storage profile information, please refer to https://aka.ms/storageprofile
          with azurerm_windows_virtual_machine.test,
          on terraform_plugin_test.tf line 78, in resource "azurerm_windows_virtual_machine" "test":
          78: resource "azurerm_windows_virtual_machine" "test" {
--- FAIL: TestAccWindowsVirtualMachine_diskOSConfidentialSecurityProfileBlock (124.19s)
FAIL

------- Stdout: -------
=== RUN   TestAccWindowsVirtualMachine_otherSecureBootEnabled
=== PAUSE TestAccWindowsVirtualMachine_otherSecureBootEnabled
=== CONT  TestAccWindowsVirtualMachine_otherSecureBootEnabled
    testcase.go:181: Step 1/1 error: Error running apply: exit status 1
        Error: creating Windows Virtual Machine (Subscription: "*******"
        Resource Group Name: "acctestRG-260724132801120197"
        Virtual Machine Name: "acctestvmjxs36"): performing CreateOrUpdate: unexpected status 400 (400 Bad Request) with error: BadRequest: Use of UEFI settings is not supported when security type is '<NULL>'.
          with azurerm_windows_virtual_machine.test,
          on terraform_plugin_test.tf line 78, in resource "azurerm_windows_virtual_machine" "test":
          78: resource "azurerm_windows_virtual_machine" "test" {
    panic.go:694: Error running post-test destroy, there may be dangling resources: exit status 1
        Error: deleting Network Interface (Subscription: "*******"
        Resource Group Name: "acctestRG-260724132801120197"
        Network Interface Name: "acctestnic-260724132801120197"): performing Delete: unexpected status 400 (400 Bad Request) with error: NicReservedForAnotherVm: Nic(s) in request is reserved for another Virtual Machine for 180 seconds. Please provide another nic(s) or retry after 180 seconds. Reserved VM: /subscriptions/*******/resourceGroups/acctestRG-260724132801120197/providers/Microsoft.Compute/virtualMachines/acctestvmjxs36
--- FAIL: TestAccWindowsVirtualMachine_otherSecureBootEnabled (92.21s)
FAIL

------- Stdout: -------
=== RUN   TestAccWindowsVirtualMachine_otherEncryptionAtHostEnabledUpdate
=== PAUSE TestAccWindowsVirtualMachine_otherEncryptionAtHostEnabledUpdate
=== CONT  TestAccWindowsVirtualMachine_otherEncryptionAtHostEnabledUpdate
    testcase.go:181: Step 1/5 error: Error running apply: exit status 1
        Error: creating Windows Virtual Machine (Subscription: "*******"
        Resource Group Name: "acctestRG-260724132800110691"
        Virtual Machine Name: "acctestvmfvw1m"): performing CreateOrUpdate: unexpected status 400 (400 Bad Request) with error: BadRequest: Use of UEFI settings is not supported for the provided image.
          with azurerm_windows_virtual_machine.test,
          on terraform_plugin_test.tf line 78, in resource "azurerm_windows_virtual_machine" "test":
          78: resource "azurerm_windows_virtual_machine" "test" {
    panic.go:694: Error running post-test destroy, there may be dangling resources: exit status 1
        Error: deleting Network Interface (Subscription: "*******"
        Resource Group Name: "acctestRG-260724132800110691"
        Network Interface Name: "acctestnic-260724132800110691"): performing Delete: unexpected status 400 (400 Bad Request) with error: NicReservedForAnotherVm: Nic(s) in request is reserved for another Virtual Machine for 180 seconds. Please provide another nic(s) or retry after 180 seconds. Reserved VM: /subscriptions/*******/resourceGroups/acctestRG-260724132800110691/providers/Microsoft.Compute/virtualMachines/acctestvmfvw1m
--- FAIL: TestAccWindowsVirtualMachine_otherEncryptionAtHostEnabledUpdate (95.60s)
FAIL

------- Stdout: -------
=== RUN   TestAccWindowsVirtualMachine_otherEncryptionAtHostEnabled
=== PAUSE TestAccWindowsVirtualMachine_otherEncryptionAtHostEnabled
=== CONT  TestAccWindowsVirtualMachine_otherEncryptionAtHostEnabled
    testcase.go:181: Step 1/1 error: Error running apply: exit status 1
        Error: creating Windows Virtual Machine (Subscription: "*******"
        Resource Group Name: "acctestRG-260724132748823640"
        Virtual Machine Name: "acctestvm9y6k4"): performing CreateOrUpdate: unexpected status 400 (400 Bad Request) with error: BadRequest: Use of UEFI settings is not supported for the provided image.
          with azurerm_windows_virtual_machine.test,
          on terraform_plugin_test.tf line 78, in resource "azurerm_windows_virtual_machine" "test":
          78: resource "azurerm_windows_virtual_machine" "test" {
--- FAIL: TestAccWindowsVirtualMachine_otherEncryptionAtHostEnabled (130.36s)
FAIL

------- Stdout: -------
=== RUN   TestAccWindowsVirtualMachine_otherVTpmEnabled
=== PAUSE TestAccWindowsVirtualMachine_otherVTpmEnabled
=== CONT  TestAccWindowsVirtualMachine_otherVTpmEnabled
    testcase.go:181: Step 1/1 error: Error running apply: exit status 1
        Error: creating Windows Virtual Machine (Subscription: "*******"
        Resource Group Name: "acctestRG-260724132917195562"
        Virtual Machine Name: "acctestvmvkust"): performing CreateOrUpdate: unexpected status 400 (400 Bad Request) with error: BadRequest: Use of UEFI settings is not supported when security type is '<NULL>'.
          with azurerm_windows_virtual_machine.test,
          on terraform_plugin_test.tf line 78, in resource "azurerm_windows_virtual_machine" "test":
          78: resource "azurerm_windows_virtual_machine" "test" {
    panic.go:694: Error running post-test destroy, there may be dangling resources: exit status 1
        Error: deleting Network Interface (Subscription: "*******"
        Resource Group Name: "acctestRG-260724132917195562"
        Network Interface Name: "acctestnic-260724132917195562"): performing Delete: unexpected status 400 (400 Bad Request) with error: NicReservedForAnotherVm: Nic(s) in request is reserved for another Virtual Machine for 180 seconds. Please provide another nic(s) or retry after 180 seconds. Reserved VM: /subscriptions/*******/resourceGroups/acctestRG-260724132917195562/providers/Microsoft.Compute/virtualMachines/acctestvmvkust
--- FAIL: TestAccWindowsVirtualMachine_otherVTpmEnabled (76.48s)
FAIL

------- Stdout: -------
=== RUN   TestAccWindowsVirtualMachine_otherEncryptionAtHostEnabledWithCMK
=== PAUSE TestAccWindowsVirtualMachine_otherEncryptionAtHostEnabledWithCMK
=== CONT  TestAccWindowsVirtualMachine_otherEncryptionAtHostEnabledWithCMK
    testcase.go:181: Step 1/1 error: Error running apply: exit status 1
        Error: creating Windows Virtual Machine (Subscription: "*******"
        Resource Group Name: "acctestRG-260724132933402207"
        Virtual Machine Name: "acctestvmaforv"): performing CreateOrUpdate: unexpected status 400 (400 Bad Request) with error: BadRequest: Use of UEFI settings is not supported for the provided image.
          with azurerm_windows_virtual_machine.test,
          on terraform_plugin_test.tf line 169, in resource "azurerm_windows_virtual_machine" "test":
         169: resource "azurerm_windows_virtual_machine" "test" {
--- FAIL: TestAccWindowsVirtualMachine_otherEncryptionAtHostEnabledWithCMK (308.24s)
FAIL

------- Stdout: -------
=== RUN   TestAccWindowsVirtualMachineScaleSet_disksOSDiskConfidentialSecurityProfileBlock
=== PAUSE TestAccWindowsVirtualMachineScaleSet_disksOSDiskConfidentialSecurityProfileBlock
=== CONT  TestAccWindowsVirtualMachineScaleSet_disksOSDiskConfidentialSecurityProfileBlock
    testcase.go:181: Step 1/3 error: Error running apply: exit status 1
        Error: creating Windows Virtual Machine Scale Set (Subscription: "*******"
        Resource Group Name: "acctestRG-VMSS-260724134511453878"
        Virtual Machine Scale Set Name: "acctvm26"): performing CreateOrUpdate: unexpected status 404 (404 Not Found) with error: PlatformImageNotFound: The platform image 'MicrosoftWindowsServer:windows-cvm:2022-datacenter-cvm:latest' is not available. Verify that all fields in the storage profile are correct. For more details about storage profile information, please refer to https://aka.ms/storageprofile
          with azurerm_windows_virtual_machine_scale_set.test,
          on terraform_plugin_test.tf line 64, in resource "azurerm_windows_virtual_machine_scale_set" "test":
          64: resource "azurerm_windows_virtual_machine_scale_set" "test" {
--- FAIL: TestAccWindowsVirtualMachineScaleSet_disksOSDiskConfidentialSecurityProfileBlock (104.74s)
FAIL
```

tamird committed:
fix: correct virtual machine security profile requests

catriona-m commented on internal/services/compute/windows_virtual_machine_resource_other_test.go:
```diff
@@ -3383,7 +3408,18 @@ resource "azurerm_windows_virtual_machine" "test" {
+`, *enabled)
+	} else if empty {
+		securityProfile = "\n  security_profile {}\n"
+	}
```
rather than setting different configs based on the presence of a bool, could we instead use different configs to achieve this? this layout is a bit difficult for maintainers to follow which config is actually being tested

tamird replied:
Replaced the nullable boolean and empty-block switch with explicit enabled, disabled, empty, and omitted HCL configurations across all four VM and scale-set lifecycle tests.

— tamirdex

catriona-m commented on internal/services/compute/virtual_machine_scale_set.go:
```diff
@@ -2240,3 +2240,98 @@ func flattenOrchestratedVirtualMachineScaleSetIdentity(input *identity.SystemAnd
+	var secureBootEnabled *bool
+	var vtpmEnabled *bool
+	securityProfile, securityProfileConfigured := d.GetOk("security_profile")
+	if securityProfileConfigured {
```
could we change this to pass in the value of `d.GetOk("security_profile")` and nil check and len check the value? otherwise this could cause a panic

tamird replied:
Both VM and scale-set expanders now receive the raw GetOk value and check its type, length, and first element before indexing. Omitted and empty values return no profile, while removing a configured profile still explicitly disables host encryption.

— tamirdex

catriona-m commented on internal/services/compute/virtual_machine_scale_set.go:
```diff
@@ -2240,3 +2240,98 @@ func flattenOrchestratedVirtualMachineScaleSetIdentity(input *identity.SystemAnd
+			return nil, fmt.Errorf("`security_profile.0.vtpm_enabled` must be set to `true` when `os_disk.0.security_encryption_type` is set")
+		}
+
+		const expected = virtualmachinescalesets.SecurityTypesConfidentialVM
```
this doesn't need to be a const, we should use the value directly from the sdk

tamird replied:
Removed both local constants; VM and scale-set validation now compare and report their respective SDK SecurityTypesConfidentialVM values directly.

— tamirdex

catriona-m reviewed (changes requested):
Thanks @tamird I left a couple more comments inline but once those are addressed I can take another look and re-run the tests. Thanks!

tamird committed:
fix: address virtual machine security profile review

tamird committed:
fix: format VM security profile test configurations

catriona-m committed:
review fixes

tamird committed:
fix: handle removed virtual machine security profiles
